### PR TITLE
IAM test for HTTP functions. Fixes #2086

### LINKED ISSUE #2086: Better IAM check and warnings when creating HTTP functions
Due to the recent change in default IAM policy for HTTP Cloud Functions, it is now the case that to make a public HTTP function the `cloudfunctions.functions.setIamPolicy` permission is required. This permission is included in `owner`, but not `editor`.

We should perform a specialized IAM check when creating an HTTP function so we can properly warn users:

1. If any of the actions to be taken by a deploy will be the creation of an HTTP function, then...
2. Run a `testIamPermissions` for the user at the project level to see if they have `cloudfunctions.functions.setIamPolicy`.
3. If they do, allow the deploy to proceed.
4. If they don't, error out with a clear error message (see below):

```
Creating HTTP functions requires permission cloudfunctions.functions.setIamPolicy which
was not detected for these credentials. This permission can be granted by 'owner' and
'cloudfunctions.admin' roles in the Firebase or Google Cloud consoles.
```

I'm on the fence about whether to make this an error or a warning. It's far from ideal in either case, honestly. I'm going to follow up separately to see if we can address this more cleanly.

### PR DESCRIPTION
**Pull request by mbleigh**
This is ugly in that the contents of `checkHttpIam.ts` are incredibly esoteric and are basically cargo-cult recreations of existing logic from `release.js`; however, I believe that this is about the cleanest implementation of this that wouldn't require a multi-day refactor of the entire functions release process.

I made a one-line change to `release.js` to re-use the results of the fetch from `deploy.ts` -- I do not intend to fix the 70+ lint errors that popped out of that.

Note that I do not intend to merge this just yet because assigning someone "Develop Admin" fixes this problem but causes another. Need to figure out the messaging, but the code should stay more or less stable.

**Update:** This PR now includes a second error check -- testing whether the account has `iam.serviceAccounts.actAs` for the App Engine default service account. Screenshots of both errors below:

actsAs error

http function error

### CODE REVIEW

File: src/deploy/functions/checkIam.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    const iamResult = await testResourceIamPermissions(
+      "https://iam.googleapis.com",
+      "v1",
+      `projects/${projectId}/serviceAccounts/${saEmail}`,
+      ["iam.serviceAccounts.actAs"]
+    );
+    passed = iamResult.passed;
+  } catch (err) {
+    debug("[functions] service account IAM check errored, deploy may fail:", err);
+    return;
```
Comment (bkendall) [inline src/deploy/functions/checkIam.ts:30]: If this return is correct, please add a comment indicating why we wouldn't want to re-throw the error.

File: src/gcp/iam.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import * as api from "../api";
 import { endpoint } from "../utils";
 import { difference } from "lodash";
+import { debug } from "winston";
```
Comment (bkendall) [inline src/gcp/iam.ts:4]: This should probably come from `../logger`, not `winston`.

File: src/deploy/functions/checkIam.ts (hunk 1 above)
Comment (mbleigh) [reply src/deploy/functions/checkIam.ts:31]: Done.

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1 @@
+- Improves error messages for Cloud Functions deployment permissions. (#2086)
```
Comment (bkendall) [inline CHANGELOG.md:1]:
```suggestion
- Improves error messages for Cloud Functions for Firebase deployment permissions. (#2086)
```

**Pull request merged**